multi-step-flow/src/app/extensions/components/OrderMealCard.tsx

```diff
@@ -0,0 +1,155 @@
+  const getRestaurants = useCallback(() => {
+    setLoading(true);
+    setError(false);
+    // Fetch a list of restaurants and their menus from our serverless function
+    hubspot
+      .fetch(RESTAURANTS_SOURCE_URL)
+      .then((response) => {
+        response.json().then((data) => {
+          // Make sure the response is the shape we expect (array of Restaurants)
+          if (Array.isArray(data)) {
+            const restaurants = data.map((restaurant: unknown) => {
+              if (
+                typeof restaurant !== 'object' ||
+                Array.isArray(restaurant) ||
+                restaurant === null
+              ) {
+                throw new TypeError('Object is not a Restaurant');
+              }
+
+              return restaurant as Restaurant;
+            });
+
+            setRestaurants(restaurants);
+            return;
+          }
+
+          throw new Error('Response is not an array.');
+        });
+      })
+      .catch((error) => {
+        console.error(error.message);
+        setError(true);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [runServerless]);
+
+  useEffect(() => {
+    getRestaurants();
+    fetchCrmObjectProperties(['firstname']).then(({ firstname }) => {
+      setContactName(firstname);
+    });
+  }, []);
+
+  const clearSelection = useCallback(() => setSelected(undefined), []);
+
+  const handleAddClick = useCallback((item: CartItem) => {
+    updateCart((items: Array<CartItem>) => [...items, item]);
+    clearSelection();
+  }, []);
+
+  const handleRemoveClick = useCallback((id: number) => {
+    updateCart((items: Array<CartItem>) =>
+      items.filter((item) => item.id !== id),
+    );
+  }, []);
+
+  const handleCheckoutClick = useCallback(
+    (message: string) => {
+      sendAlert({
+        type: 'success',
+        message: `Nicely done! The meal is on its way to ${contactName} with the message: "${message}"`,
+      });
+      updateCart([]);
+      clearSelection();
+    },
+    [contactName],
+  );
+
+  if (error) {
+    return (
+      <ErrorState title="Unable to fetch restaurants">
+        <Text>Please wait a moment and try again.</Text>
+        <Button onClick={getRestaurants}>Retry</Button>
+      </ErrorState>
+    );
+  }
+
+  if (loading) {
+    return (
+      <LoadingSpinner layout="centered" label="Loading..." showLabel={true} />
+    );
+  }
+
+  // Small utility function for help below
+  const getRestaurant = (id?: number) => {
+    return restaurants.find((r) => r.id === id);
+  };
+
+  const uniqueRestaurants = new Set(cart.map((item) => item.restaurantId));
+  const totalDeliveryCost = [...uniqueRestaurants].reduce((total, id) => {
+    return total + (getRestaurant(id)?.deliveryCost ?? 0);
+  }, 0);
+
+  const selectedRestaurant = getRestaurant(selected);
+  const subtotal = cart.reduce((total, item) => total + item.price, 0);
+
+  return (
+    <Flex direction={'column'} gap={'md'}>
```

Comment:
Super nit, but these can be `direction="column" gap="md"`. There's a couple other instances of this around.